src/ripple/app/paths/PathRequest.cpp

```diff
@@ -279,65 +267,87 @@ Json::Value PathRequest::doCreate (
+    {
+        jvStatus = rpcError(rpcSRC_ACT_MISSING);
+        return PFR_PJ_INVALID;
+    }
 
-    if (jvParams.isMember (jss::source_account))
+    if (! jvParams.isMember(jss::destination_account))
     {
-        raSrcAccount = parseBase58<AccountID>(
-            jvParams[jss::source_account].asString());
-        if (! raSrcAccount)
-        {
-            jvStatus = rpcError (rpcSRC_ACT_MALFORMED);
-            return PFR_PJ_INVALID;
-        }
+        jvStatus = rpcError(rpcDST_ACT_MISSING);
+        return PFR_PJ_INVALID;
     }
-    else if (complete)
+
+    if (! jvParams.isMember(jss::destination_amount))
     {
-        jvStatus = rpcError (rpcSRC_ACT_MISSING);
+        jvStatus = rpcError(rpcDST_AMT_MISSING);
         return PFR_PJ_INVALID;
     }
 
-    if (jvParams.isMember (jss::destination_account))
+    raSrcAccount = parseBase58<AccountID>(
+        jvParams[jss::source_account].asString());
+    if (! raSrcAccount)
     {
-        raDstAccount = parseBase58<AccountID>(
-            jvParams[jss::destination_account].asString());
-        if (! raDstAccount)
-        {
-            jvStatus = rpcError (rpcDST_ACT_MALFORMED);
-            return PFR_PJ_INVALID;
-        }
+        jvStatus = rpcError (rpcSRC_ACT_MALFORMED);
+        return PFR_PJ_INVALID;
     }
-    else if (complete)
+
+    raDstAccount = parseBase58<AccountID>(
+        jvParams[jss::destination_account].asString());
+    if (! raDstAccount)
     {
-        jvStatus = rpcError (rpcDST_ACT_MISSING);
+        jvStatus = rpcError (rpcDST_ACT_MALFORMED);
         return PFR_PJ_INVALID;
     }
 
-    if (jvParams.isMember (jss::destination_amount))
+    if (! amountFromJsonNoThrow (
+        saDstAmount, jvParams[jss::destination_amount]))
     {
-        if (! amountFromJsonNoThrow (
-                saDstAmount, jvParams[jss::destination_amount]) ||
-            (saDstAmount.getCurrency ().isZero () &&
-             saDstAmount.getIssuer ().isNonZero ()) ||
-            (saDstAmount.getCurrency () == badCurrency ()) ||
-            saDstAmount <= zero)
-        {
-            jvStatus = rpcError (rpcDST_AMT_MALFORMED);
-            return PFR_PJ_INVALID;
-        }
+        jvStatus = rpcError (rpcDST_AMT_MALFORMED);
+        return PFR_PJ_INVALID;
     }
-    else if (complete)
+
+    convert_all_ =
+        saDstAmount == STAmount(saDstAmount.issue(), 1u, 0, true);
+
+    if ((saDstAmount.getCurrency ().isZero () &&
+            saDstAmount.getIssuer ().isNonZero ()) ||
+        (saDstAmount.getCurrency () == badCurrency ()) ||
+        (! convert_all_ && saDstAmount <= zero))
     {
-        jvStatus = rpcError (rpcDST_ACT_MISSING);
+        jvStatus = rpcError (rpcDST_AMT_MALFORMED);
         return PFR_PJ_INVALID;
     }
 
+    if (jvParams.isMember(jss::send_max))
+    {
+        // Send_max requires destination amount to be -1.
+        if (! convert_all_)
+        {
+            jvStatus = rpcError(rpcDST_ACT_MISSING);
```

Comment:
I think this should be `rpcDST_AMT_MALFORMED`, or maybe `rpcDST_AMT_MISSING`. But definitely not anything related to `DST_ACT`.